EleksTubeHAX_pio/src/TFTs.cpp

```diff
@@ -35,25 +41,35 @@ void TFTs::begin() {
 }
 
 void TFTs::reinit() {
-  #ifndef TFT_SKIP_REINIT
+#ifndef TFT_SKIP_REINIT
   // Start with all displays selected.
   chip_select.begin();
   chip_select.setAll();
 
-  // Turn power on to displays.
-  pinMode(TFT_ENABLE_PIN, OUTPUT);
-  #ifdef DIM_WITH_ENABLE_PIN_PWM
+#ifdef DIM_WITH_ENABLE_PIN_PWM
   ledcAttachPin(TFT_ENABLE_PIN, TFT_PWM_CHANNEL);
   ledcChangeFrequency(TFT_PWM_CHANNEL, 20000, 8);
-  ProcessUpdatedDimming();
-  #endif
-  enableAllDisplays();
+  //same thing as on first init, not needed here, because enableAllDisplays() is called later and it is called from there and if enabled is false, this does nothing
+  //ProcessUpdatedDimming();
+#else
+  // Turn power on to displays.
+  pinMode(TFT_ENABLE_PIN, OUTPUT);
+#endif
+  // Signal, that the image in the buffer is invalid and needs to be reloaded and refilled
+  // needed, because the last drawing can be long time ago?!?
+  InvalidateImageInBuffer();
 
-  // Initialize the super class.
+  // Initialize the super class (again).
   init();
-  #else
+  //to avoid flickering patterns on the screens
+  fillScreen(TFT_BLACK);
```

Comment:
Maybe need to think whether `enableAllDisplays` must be called before this or not.
On IPSTUBE with HW dimming doing this call before that function will result in black screen when activated - OK.
But for other clocks might it be required to call `enableAllDisplays` in advance to deliver command to them?

Reply:
I will try it out ;)

Reply:
I played around a lot with the activation and reactivation sequences...Found a way, that is working for all clocks I own...I don't own a "real" Elekstube Gen1, so not sure, if everthing is working with it. The CyberPunk clock is VERY close to the orginal (just reversed display sequence), so for me, everthing looks goos now.